src/main/java/org/myrobotlab/framework/Service.java

```diff
@@ -782,14 +782,18 @@ public String getRootDataDir() {
   public String getHomeDir() {
     return System.getProperty("user.home");
   }
-
-  public String getDataDir() {
-    String dataDir = Runtime.getOptions().dataDir + fs + getClass().getSimpleName();
+  
+  static public String getDataDir(String typeName) {

```

Comment:
Why does this method need a typeName passed into it now?  Implicitly, the service should already know what type it is.